pkg/repo/util.go

```diff
@@ -17,39 +16,31 @@ import (
 
 // initRepoFiles initializes all files required for a valid bacalhau repo.
 func initRepoFiles(cfg types.BacalhauConfig) error {
-	if err := initUserIDKey(cfg.User.KeyPath); err != nil {
+	if err := initUserIDKey(cfg.UserKeyPath()); err != nil {
 		return fmt.Errorf("failed to create user key: %w", err)
 	}
-
-	if err := initDir(cfg.Node.ExecutorPluginPath); err != nil {
+	if err := ensureDir(cfg.EnginePluginsDir()); err != nil {
 		return fmt.Errorf("failed to create plugin dir: %w", err)
 	}
-
-	if err := initDir(cfg.Node.ComputeStoragePath); err != nil {
-		return fmt.Errorf("failed to create executor storage dir: %w", err)
+	if err := ensureDir(cfg.OrchestratorDir()); err != nil {
+		return fmt.Errorf("failed to create orchestrator dir: %w", err)
 	}
-	if err := initParentDir(cfg.Node.Compute.ExecutionStore.Path); err != nil {
-		return fmt.Errorf("failed to create executor execution store dir: %w", err)
+	if err := ensureDir(cfg.ComputeDir()); err != nil {
+		return fmt.Errorf("failed to create executor storage dir: %w", err)
 	}
-	if err := initParentDir(cfg.Node.Requester.JobStore.Path); err != nil {
-		return fmt.Errorf("failed to create orchestrator job store dir: %w", err)
+	if err := ensureDir(cfg.ExecutionDir()); err != nil {
+		return fmt.Errorf("failed to create executor storage dir: %w", err)
 	}
```

Comment:
You are still creating unnecessary directories. Why do we still need to create `ComputeDir` and `ExecutionDir` if the user is running an orchestrator node or just running a client? I have asked multiple times to initialize dirs in the getters until we move the logic in `node` package in the future if we see a need for that. Please lets stop wasting more time on this and take this as a requirement. Do initialize the dirs in the getters and lets move forward

Reply:
fixed in latest changes